src/main/java/com/icthh/xm/tmf/ms/communication/service/FirebaseService.java

```diff
@@ -0,0 +1,60 @@
+package com.icthh.xm.tmf.ms.communication.service;
+
+import com.icthh.xm.tmf.ms.communication.domain.FirebaseRequest;
+import com.icthh.xm.tmf.ms.communication.domain.FirebaseRequestData;
+import com.icthh.xm.tmf.ms.communication.utils.ApiMapper;
+import com.icthh.xm.tmf.ms.communication.web.api.model.Receiver;
+import com.icthh.xm.tmf.ms.communication.web.rest.errors.FirebaseCommunicatoinException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+@Service
+public class FirebaseService {
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${application.firebase.url}")
+    private String firebaseURL;
+
+    @Value("${application.firebase.token}")
+    private String token;
```

Comment:
better to move to configuration on configuration ms
maybe add microservice configuration like uaa.yml but for this ms

Reply:
will be provided as separate PR for configuration MS

Reply:
configuration is re-implemented